source/guide_codimd.rst

```diff
@@ -0,0 +1,398 @@
+
+.. sidebar:: Logo
+
+  .. image:: _static/images/codimd.png
+      :align: center
+
+######
+CodiMD
+######
+
+CodiMD_ lets you create real-time collaborative markdown notes on all platforms.
+Inspired by Hackpad, with more focus on speed and flexibility, and build from HackMD_ source code.
+
+----
+
+.. note:: For this guide you should be familiar with the basic concepts of
+
+  * MySQL_
+  * supervisord_
+  * domains_
+  * Node_ and its package manager npm_
+
+
+Prerequisites
+=============
+
+.. include:: includes/my-print-defaults.rst
+
+We're using Node_ in the stable version 8:
+
+::
+
+ [isabell@stardust ~]$ uberspace tools version show node
+ Using 'Node.js' version: '8'
+ [isabell@stardust ~]$
+
+Your blog URL needs to be setup:
+
+.. include:: includes/web-domain-list.rst
+
+Installation
+============
+
+You need a special database for CodiMD_:
+
+.. warning:: Replace ``<username>`` with your username!
+
+Create Database
+---------------
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ mysql -e "CREATE DATABASE <username>_codimd CHARACTER SET utf8 COLLATE utf8_general_ci;"
+  [isabell@stardust ~]$
+
+Download Sources
+----------------
+
+go to the release_-site and download the latest version 
+
+.. code-block:: console
+
+  [isabell@stardust ~]$ wget https://github.com/hackmdio/codimd/archive/6.6.66.zip
+  --2018-10-17 18:50:29--  https://github.com/hackmdio/codimd/archive/6.6.66.zip
+  Resolving github.com (github.com)... 192.30.253.112, 192.30.253.113
+  Connecting to github.com (github.com)|192.30.253.112|:443... connected.
+  HTTP request sent, awaiting response... 302 Found
+  Location: https://codeload.github.com/hackmdio/codimd/zip/1.2.1 [following]
+  --2018-10-17 18:50:29--  https://codeload.github.com/hackmdio/codimd/zip/1.2.1
+  Resolving codeload.github.com (codeload.github.com)... 192.30.253.121, 192.30.253.120
+  Connecting to codeload.github.com (codeload.github.com)|192.30.253.121|:443... connected.
+  HTTP request sent, awaiting response... 200 OK
+  Length: unspecified [application/zip]
+  Saving to: ‘6.6.66.zip’
+  
+      [     <=>                                                                                                                                        ] 7,090,453   6.28MB/s   in 1.1s   
+  
+  2018-10-17 18:50:31 (6.28 MB/s) - ‘6.6.66.zip’ saved [7090453]
+  [isabell@stardust ~]$ unzip 6.6.66.zip
+  [...]
+  [isabell@stardust ~]$ mv codimd-6.6.66/ codimd
+  [isabell@stardust ~]$ rm 6.6.66.zip
+  [isabell@stardust ~]$ cd codimd
+  [isabell@stardust codimd]$ bin/setup
+  copy config files
+  install npm packages
+  [...]
+  [isabell@stardust codimd]$
+
+
+Remove UglifyJS
+---------------
+
+The UglifyJS plugin freezes the compilation step, because it tries to compile everything in parallel and then reaches the maximum memory-limit/user of uberspace. And since it's apparently not necessary, we'll just remove it.
+
+Open ``webpack.production.js`` and comment out this line (currently line 6):
```

Comment:
Please explain why

Reply:
compiling everything in parallel uses more ram.
the available ram per user at uberspace is around ~1,5GB.
the parallel compilation needs more than 2GB.
the sequential compilation needs less than 1,5GB.
that is why we will use the sequential one.
CodiMD itself uses parallel by default, but we can disable it via uncommenting `UglifyJS`